src/app/shared/services/json-utils.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable } from '@angular/core';
+import { SchemaKeysStoreService } from './schema-keys-store.service';
+import * as _ from 'lodash';
+
+@Injectable()
+export class JsonUtilsService {
+
+  constructor(public schemaKeysStoreService: SchemaKeysStoreService) { }
+
+  filterJsonArray(jsonArray: Array<object>, filterExpression: string) {
+    let _jsonArray = [];
+    jsonArray.forEach(record => {
+      // cast to array for now before adding multiple filter expressions
+      let res = this.filterJson(record, [filterExpression]);
+      if (!_.isEmpty(res)) {
+        _jsonArray.push(res);
+      }
+    });
+    return _jsonArray;
+  }
+
+  filterJson(json: object, paths: Array<string>): object {
+    let result = {};
+    paths.forEach(path => {
+      let res = this.filterJsonRecursively(Object.assign({}, json), path);
+      if (res) {
+        _.merge(result, res);
+      }
+    });
+    this.cleanup(result);
```

Comment:
I am not sure why you need to clean the values and from where this undefined values coming exactly but you can maybe do try to do cleanup, during `merge` via `mergeWith` in order to avoid second traverse of whole record.

Reply:
I want to clean them because the initial implementation is with multiple tags. So, even if in one tag there are some undefined values when the filiter applied and didn't match maybe will match in another tag and then it's gonna merge correctly. There was a problem in merging arrays. If I clean them on merge the correct order is lost for the elements.